nucypher-core/src/dkg.rs

```diff
@@ -0,0 +1,115 @@
+use alloc::boxed::Box;
+use alloc::string::String;
+
+use serde::{Deserialize, Serialize};
+
+use crate::conditions::{Conditions, Context};
+
+use crate::versioning::{
+    messagepack_deserialize, messagepack_serialize, ProtocolObject, ProtocolObjectInner,
+};
+
+/// The ferveo variant to use for the decryption share derivation.
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize, Clone)]
+pub enum FerveoVariant {
+    /// the simple variant requires n/n shares to decrypt
+    SIMPLE,
+    /// the precomputed variant requires m/n shares to decrypt
+    PRECOMPUTED,
+}
+
+/// A request for an Ursula to derive a decryption share.
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize)]
+pub struct ThresholdDecryptionRequest {
+    /// The ID of the ritual.
+    pub ritual_id: u16,
+    /// The ciphertext to generate a decryption share for.
+    pub ciphertext: Box<[u8]>,
```

Comment:
Add `#[serde(with = "serde_bytes::as_base64")]` (meaning that it will be serialized as a bytestring for binary formats, and as base64 in human-readable ones), or it'll be serialized as a vector of integers instead.

Reply:
✔️